### DISCUSSION TITLE
"[@types/node-forge]  Missing setAttributes() in CertificateRequest

### DISCUSSION THREAD

**Discussion by Ichnafi** [DefinitelyTyped/DefinitelyTyped#66973] [Issues with a @types package]
As mentioned in an [issue 1036](https://github.com/digitalbazaar/forge/issues/1036) in the node-forge github, there is a definition for setAttributes() missing, so one can not create CSR with subject alternative names.
The only declaration existing is for getAttributes() in index.d.ts linte 562.

Since I'm no proffecional programmer, I have no idea how to fix this, but node-forge ist very important for me.

![missing-setAttribute](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/26506940/127df984-02bb-4595-8d9f-c4ba9e724d9b)

Changing the interface for CertificateRequest to something like this, seems to do the trick, but I'm not sure if this is even remotly correct in any share or form.
```
/**
 * Sets attributes of this certificate request.
 *
 * @param attrs the array of subject attributes to use.
 */
setAttributes(attr: CertificateField[]): Attribute | null;
```
Can anyone confirm?

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66985
